### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by whiteshadow-gr**
* Refactored the option collector slightly
* changed scala and sbt version

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -69,6 +69,7 @@ val dataplug = project
       Library.Specs2.core,
       Library.Specs2.matcherExtra,
       Library.Specs2.mock,
+      Library.Utils.apacheCommonLang,
```
Comment (benshaw) [inline build.sbt:72]: why do we need this ?

File: dataplug/app/com/hubofallthings/dataplug/apiInterfaces/DataPlugOptionsCollector.scala [hunk 2]
```diff
@@ -13,38 +13,47 @@ import com.hubofallthings.dataplug.apiInterfaces.authProviders.RequestAuthentica
...
+        error(s"Error when querying api endpoint $fetchParams - ${e.getMessage}", e)
+    }
+  }
+
+  def error(errorMessage: String, exception: Throwable): Future[Nothing] = {
+    logger.warn(errorMessage)
+    Future.failed(exception)
+  }
+
+  def unauthorizedResponse(fetchParams: ApiEndpointCall, hatAddress: String, hatClientActor: ActorRef, retrying: Boolean, response: WSResponse)
```
Comment (benshaw) [inline dataplug/app/com/hubofallthings/dataplug/apiInterfaces/DataPlugOptionsCollector.scala:47]: Can this be private or even better local to get method ?

File: build.sbt (hunk 1 above)
Comment (whiteshadow-gr) [reply build.sbt:72]: This is needed cause Java 11. It's what solved the compatibilities we had before. That means that the ugliness with the reflection remains

Comment (benshaw) [reply build.sbt:72]: so just adding this import fixes it ?

Comment (whiteshadow-gr) [reply build.sbt:72]: Yes 👍

File: project/Dependencies.scala [hunk 3]
```diff
@@ -76,6 +76,7 @@ object Dependencies {
 
     object Utils {
       val akkaHttp = "com.typesafe.akka" %% "akka-http" % "10.0.9"
+      val apacheCommonLang = "org.apache.commons" % "commons-lang3" % "3.10"
```
Comment (whiteshadow-gr) [inline project/Dependencies.scala:79]: @benshaw here is the version

File: build.sbt (hunk 1 above)
Comment (whiteshadow-gr) [reply build.sbt:72]: The incompatibility is in this package. It turns out they have pushed an update that solves the errors. Gus found this solution

**Pull request merged**